test/pagination/services.ts

```diff
@@ -0,0 +1,85 @@
+import { strict as assert } from 'node:assert';
+import { after, before, describe, it } from 'node:test';
+
+import * as dotenv from 'dotenv';
+dotenv.config({ path: './.env.test'});
+
+import { PrincipalService, recordToModel } from '../../src/principal/service.ts';
+import db, { insertAndGetId } from '../../src/database.ts';
+
+describe('findAll users pagination', () => {
+
+  const userRecordsDb: unknown[] = [];
+  before(async () => {
+
+    await db.schema.createTable('principals', (table) => {
+      table.increments('id').primary();
+      table.string('external_id').notNullable();
+      table.string('nickname').notNullable();
+      table.integer('type').notNullable();
+      table.boolean('active').notNullable();
+      table.timestamp('created_at').defaultTo(db.fn.now());
+      table.timestamp('modified_at').defaultTo(db.fn.now());
+      table.boolean('system').notNullable();
+    });
```

Comment:
Could we run all migrations on the test database when it starts? This ensures that tests break if the schema changes.